### DISCUSSION TITLE
Rating w/ Font Awesome Icon

### DISCUSSION THREAD

**Discussion by MarcHurst** [saadeghi/daisyui#1557] [Q&A] [answered]
Hey guys...

I'm pretty terrible with CSS... which is why this has been so great.

I've gotten the rating working on a page in an app.  When I pick a rating, the others following it dim out, as expected.

Then I changed the mask-heart class to a font-awesome icon...  I had to change the bg colour to a text colour to have the colour move across... but by doing so, it doesn't.  If I change it back to bg-color it works, but it doesn't colour things the same.

So I guess the question is... is there a way to make this work that I'm missing?

```
    <div class='rating rating-lg'>
      <label class='label'>
        <span class='label-text'>Rating</span>
      </label>
      {{{reportRating report.rating}}}
    </div>
```

Which calls the helper function:
```
  reportRating: (rating) => {
    let returnHTML = `<input type='radio' name='rating' required value=0 class='rating-hidden pt-10' />`
    for (let i = 0; i <= 4; i++) {
      returnHTML += `<input type='radio' name='rating' required value=${i + 1} class='${colortable[i]} mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100' ${i + 1 == rating ? `checked` : ''}/>`
    }
    return returnHTML
  },
```

This renders out to 

```
<div class="rating rating-lg">
      <label class="label">
        <span class="label-text">Rating</span>
      </label>
      <input type="radio" name="rating" required="" value="0" class="rating-hidden pt-10">
<input type="radio" name="rating" required="" value="1" class="text-red-400 mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100">
<input type="radio" name="rating" required="" value="2" class="text-orange-400 mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100">
<input type="radio" name="rating" required="" value="3" class="text-yellow-400 mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100">
<input type="radio" name="rating" required="" value="4" class="bg-lime-400 mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100">
<input type="radio" name="rating" required="" value="5" class="bg-green-400 mask fa-solid fa-cannabis fa-xl pt-10 bg-base-100">
    </div>
```

(I did 3 with text, 2 with bg to troubleshoot and see the difference).

![Screen Shot 2023-01-25 at 9 12 34 AM](https://user-images.githubusercontent.com/97694451/214585924-2abe04f1-ab2c-41ca-8d1d-d11866255a4e.png)

https://user-images.githubusercontent.com/97694451/214585741-6a19db7d-9805-4600-bd9d-810774a609f2.mov

**Comment by saadeghi** [comment, marked as answer]
It's a little challenging because daisyUI uses CSS mask to shape the icons but Font Awesome inserts icons using `:before`.  
But it is possible using CSS `background-clip`.  
Here's an example:
```html
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.2.1/css/all.min.css"/>

<div class="rating rating-lg">
  <input type="radio" name="rating" required="" value="0" class="rating-hidden">
  <input type="radio" name="rating" required="" value="1" class="bg-red-400 px-2 py-4 fa-solid fa-cannabis fa-xl bg-clip-text [-webkit-text-fill-color:transparent]">
  <input type="radio" name="rating" required="" value="2" class="bg-orange-400 px-2 py-4 fa-solid fa-cannabis fa-xl bg-clip-text [-webkit-text-fill-color:transparent]">
  <input type="radio" name="rating" required="" value="3" class="bg-yellow-400 px-2 py-4 fa-solid fa-cannabis fa-xl bg-clip-text [-webkit-text-fill-color:transparent]">
  <input type="radio" name="rating" required="" value="4" class="bg-lime-400 px-2 py-4 fa-solid fa-cannabis fa-xl bg-clip-text [-webkit-text-fill-color:transparent]">
  <input type="radio" name="rating" required="" value="5" class="bg-green-400 px-2 py-4 fa-solid fa-cannabis fa-xl bg-clip-text [-webkit-text-fill-color:transparent]">
</div>
```

https://play.tailwindcss.com/BKGHr3unSQ

**Reply by MarcHurst** [reply]
Thank you SO much!
Also for showing me that triple tick works similar to discord... so that's handy too!

So to understand: the bg-clip-text [] will add the transparency to the text colour ... cascading over what that pesky font messed up at the start?

**Reply by saadeghi** [reply]
When we want to clip the background to the shape of text in front of it (Font awesome is basically `text`) we should use this CSS:
```css
background-clip: text;
-webkit-text-fill-color: transparent;
```
The first one has a Tailwind CSS class: `bg-clip-text` but for the second one I used an arbitrary class: `[-webkit-text-fill-color:transparent]`
